File: Arrays/remove-duplicates/removeDuplicates.go
package main

import "fmt"

func removeDuplicates(arr []int) []int {
	if len(arr) == 0 {
		return []int{}
	}
	writeIndex := 1

	for readIndex := 1; readIndex < len(arr); readIndex++ {
		if arr[readIndex] != arr[readIndex-1] {
			arr[writeIndex] = arr[readIndex]
			writeIndex++
		}
	}
	
	return arr[:writeIndex]
}
func main() {
	sortedArray := []int{1, 1, 1, 2, 3, 3, 4, 5, 5}
	sortedArray = removeDuplicates(sortedArray)
	fmt.Println(sortedArray)
}

File: Arrays/remove-duplicates/removeDuplicates.js
const removeDuplicates = (arr) => {
  if (arr.length === 0) return;
  let writeIndex = 1;

  for (readIndex = 1; readIndex < arr.length; readIndex++) {
    if (arr[readIndex] !== arr[readIndex - 1]) {
      arr[writeIndex] = arr[readIndex];
      writeIndex++;
    }
  }

  arr.length = writeIndex;
  return writeIndex;
};

const sortedArr = [1, 1, 2, 2, 3, 4, 4, 5]
removeDuplicates(sortedArr);
console.log(sortedArr)